pos_payment_terminal/static/tests/pos_payment_terminal_tests.js

```diff
@@ -0,0 +1,272 @@
+                                    reject();
+                                })
+                            );
+                        }
+                        console.log("Unexpected call, ignoring");
+                    };
+                    await assert.rejects(this.payment_terminal.send_payment_request());
+                }
+            );
+
+            QUnit.test(
+                "Sad flow 2 - Slow failure/rejection: timeout in PIN machine response",
+                async function (assert) {
+                    assert.expect(7);
+                    var self = this;
+
+                    self.env.pos.proxy.connection.rpc = async function (
+                        url,
+                        params,
+                        options
+                    ) {
+                        console.log("MOCK CALL TO", url, params, options);
+                        // We get back a transaction ID, that we can then poll for
+                        if (url === "/hw_proxy/payment_terminal_transaction_start") {
+                            return new Promise((resolve) => {
+                                resolve({transaction_id: "X1234"});
+                            });
+                        } else if (url === "/hw_proxy/status_json") {
+                            return promiseTimeout(
+                                options.timeout,
+                                new Promise(async (resolve) => {
+                                    // Too long! > 1000 ms
+                                    await new Promise((r) => setTimeout(r, 10000));
+                                    resolve({
+                                        test_driver: {
+                                            is_terminal: true,
+                                            transactions: [
+                                                {
+                                                    transaction_id: "X1234",
+                                                    success: true,
+                                                    status: "SUCCESS",
+                                                    status_details: "Pretty successful",
+                                                },
+                                            ],
+                                        },
+                                    });
+                                })
+                            );
+                        }
+                        console.log("Unexpected call, ignoring");
+                    };
+                    assert.rejects(self.payment_terminal.send_payment_request());
+                }
+            );
+
+            QUnit.test(
+                "Sad flow 3 - transaction result never comes (maybe PIN did never receive)",
+                async function (assert) {
+                    assert.expect(7);
+                    var self = this;
+
+                    self.env.pos.proxy.connection.rpc = async function (
```

Comment:
It would be nice if part of this function could be shared between all the tests to reduce code